### PR TITLE
add potential fix for which ails

### PR DESCRIPTION
**Pull request by gforsyth**
a fix for those things `which` ails you?  I hope?  Maybe?

ping @scopatz @Carreau

### CODE REVIEW

File: xonsh/aliases.py [hunk 1]
```diff
@@ -329,12 +329,15 @@ def bang_bang(args, stdin=None):
 def which_version():
     """Returns output from system `which -v`"""
     if sys.platform == 'darwin':
-        if ('gnu' not in subprocess.check_output(['which'],
-                                                 stderr=subprocess.PIPE)):
+        try:
+            subprocess.check_output(['which', '-v'],
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE).decode('UTF-8')
+        except subprocess.CalledProcessError
```
Comment (Carreau) [inline xonsh/aliases.py]: end with `:`

Comment (Carreau) [inline xonsh/aliases.py]: ValueError: stdout argument not allowed, it will be overridden.

**Comment by gforsyth** [comment]
Thanks @Carreau -- I'm not at my finest at the moment.  

Does this work for you?  I'd prefer your solution but we'd have to give up Python 3.4 support.

File: xonsh/aliases.py [hunk 2]
```diff
@@ -329,12 +329,14 @@ def bang_bang(args, stdin=None):
 def which_version():
     """Returns output from system `which -v`"""
     if sys.platform == 'darwin':
-        if ('gnu' not in subprocess.check_output(['which'],
-                                                 stderr=subprocess.PIPE)):
+        try:
```
Comment (scopatz) [inline xonsh/aliases.py]: I think that this try-except block should be bumped up to the top level. I am on linux (Ubuntu) but I also don't have a `-v` option:

```console
scopatz@artemis ~ $ /usr/bin/which --version
Illegal option --
Usage: /usr/bin/which [-a] args
```

**Comment by Carreau** [comment]
> I'd prefer your solution but we'd have to give up Python 3.4 support.

Ah... sorry, 

> I'd prefer your solution but we'd have to give up Python 3.4 support.

Why ? Subprocess run was already used before my patch.

**Comment by scopatz** [comment]
Also note that I think you can pass `stderr=subprocess.STDOUT` to pipe everything to stdout.

File: xonsh/aliases.py (hunk 2 above)
Comment (gforsyth) [inline xonsh/aliases.py]: THE WHOLE WORLD IS UPSIDE-DOWN.  On it.

**Comment by Carreau** [comment]
> Also note that I think you can pass stderr=subprocess.STDOUT to pipe everything to stdout.

`which` seem too look like the dark knight.

File: xonsh/aliases.py [hunk 3]
```diff
@@ -329,12 +329,14 @@ def bang_bang(args, stdin=None):
...
+            subprocess.check_output(['which', '-v'],
+                                    stderr=subprocess.PIPE).decode('UTF-8')
+        except subprocess.CalledProcessError:
             return '<no version number available on OS X>'
```
Comment (Carreau) [inline xonsh/aliases.py]: remove `on OS X`

**Comment by Carreau** [comment]
Otherwise works for me.

**Comment by gforsyth** [comment]
I may have cut off both of `which` arms and legs now?

**Comment by Carreau** [comment]
> I may have cut off both of which arms and legs now?

I guess so.

**Comment by scopatz** [comment]
Thanks @gforsyth!

**Pull request merged**